server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -536,8 +570,21 @@ public ReplicationStats execute(Client client) throws Exception {
 
       // If we have some edits to send
       if (0 < edits.walEdits.getEditsSize()) {
+        // Check if we are interrupted before to writing the edits
+        if (Thread.interrupted()) {
+          log.debug("Replication work interrupted before writing edits, returning empty replication stats");
+          return new ReplicationStats(0L, 0L, 0L);
+        }
+
         log.debug("Sending {} edits", edits.walEdits.getEditsSize());
         long entriesReplicated = client.replicateLog(remoteTableId, edits.walEdits, tcreds);
```

Comment:
> Digging in a bit I see that ReplicationClient#getServicerConnection is using ThriftUtil#getClientNoTimeout? Maybe that is all that needs to change and this entire thing is moot.

That might actually be a much more simple solution than what you've put together. Sorry I didn't think to suggest that.